lib/matplotlib/cbook/__init__.py

```diff
@@ -1703,7 +1706,22 @@ def safe_first_element(obj):
             pass
         raise RuntimeError("matplotlib does not support generators "
                            "as input")
-    return next(iter(obj))
+
+
+def _safe_first_non_none(obj, skip_none=False):
```

Comment:
```suggestion
def _safe_first_non_none(obj, skip_none=True):
```

Reply:
Only from mobile, so maybe I’m missing some context, but why do we have two methods now? Can’t we integrate the logic easily in the original method, that now grew the skip_none parameter already? Also, having a method `safe_first_non_none()` that is configurable via `skip_none` seems contradictory in itself.

Reply:
@timhoffm 
I created the `_safe_first_non_none()` as a helper function. I guess we need to use kwargs as well  if we are integrating the logic in  `safe_first_element()`?